e2e/run-e2e-test

```diff
@@ -1,13 +1,25 @@
 #!/bin/bash
 #
 # Script to run Playwright tests with a specified app name.
-# Requires the APP_NAME environment variable to be set.
+# Requires the APP_NAME and BASE_URL environment variables to be set.
 
 # Ensure APP_NAME is provided
 if [[ -z "${APP_NAME}" ]]; then
   echo "You must pass in a specific APP_NAME. IE: APP_NAME=app npm run e2e-test" >&2
   exit 1
 fi
 
-# Run Playwright tests with the specified app name.
-npx playwright test --config "${APP_NAME}/playwright.config.js"
+# Ensure BASE_URL is provided
+if [[ -z "${BASE_URL}" ]]; then
+  echo "You must pass in a BASE_URL. IE: BASE_URL=http://localhost:3001 npm run e2e-test" >&2
+  exit 1
+fi
+
+# Check if UI is true and run Playwright UI mode
+if [[ "${UI}" == "true" ]]; then
+  echo "Running Playwright UI mode with APP_NAME=${APP_NAME}, BASE_URL=${BASE_URL}"
+  npx playwright test --config "${APP_NAME}/playwright.config.js" --ui
+else
+  echo "Running Playwright tests with APP_NAME=${APP_NAME}, BASE_URL=${BASE_URL}"
+  npx playwright test --config "${APP_NAME}/playwright.config.js"
+fi
```

Comment:
this part feels a bit clunky. @doshitan could you take a look at this PR to see if you have any ideas for how to make this part cleaner? the gist of the design is:

- make targets for running e2e suite either natively or in docker
- npm targets for abstracting playwright commands into a set of e2e scripts or calling the e2e/run-e2e-test script for actually running
- e2e/run-e2e-test script for running playwright tests and for checking some required arguments. in particular, the need for the BASE_URL env var isn't obvious since that isn't used in the command to playwright but rather it's used in playwright.config.js

my thoughts:
- it would be nice to be able to have a catchall argument to the make target to be able to pass additional args to the underlying playwright command
- it would be nice to do param checks cleanly in package.json directly without the need for the run-e2e-test wrapper but i don't know of a simple way to do that, and it's nice that this is explicit and doesn't have additional dependencies on npm libraries
- it would be nice to not have to re-implement the --ui parameter to playwright as a UI environment variable to run-e2e-test wrapper script since that makes run-e2e-test less like a simple wrapper and more like it's own abstraction

thoughts?

Reply:
Hmm, I'll have to take a deeper look, but `bin/run-e2e-test` could at minimum pass along any args passed to it, e.g.:

```bash
npx playwright test --config "${APP_NAME}/playwright.config.js" "$@"
```

So can remove the `UI` env var, just have `e2e-test:ui` run `./run-e2e-test --ui` (or drop `e2e-test:ui` and just instruct folks to `npm run e2e-test -- --ui`. And similar for the make targets, pass along stuff from a generic `args` to the `npm run` call.

I don't know about the `BASE_URL` env var requirement. Isn't that specified in the `playwright.config.js` file for each app already?

Reply:
> I don't know about the BASE_URL env var requirement. Isn't that specified in the playwright.config.js file for each app already?

good point. @rylew1 why do we even need to require `BASE_URL`? seems like that can be an optional env var that's only used in CI but not needed when running locally.

Reply:
>seems like that can be an optional env var 
Yea I think it can be optional with specified defaults in each app's playwright config. However just testing locally with multiple projects open, I'm always kicked over to port 3001 running `platform-test-nextjs` - so I always pass it in - maybe something like this then:
```
# Inform about optional BASE_URL usage
if [[ -z "${BASE_URL}" ]]; then
  echo "Warning: BASE_URL not specified. Using default configuration in playwright.config.js"
fi
```